### PR TITLE
Issue 48509: Obfuscate legitimate JavaScript and SQL content to avoid rejection

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
Web application firewalls have rules that protect against SQL and script injection. For 99.99% of all parameters, that's great. But we have a few endpoints that accept SQL and script.

#### Related Pull Requests
- https://github.com/LabKey/labkey-api-js/pull/159
- https://github.com/LabKey/testAutomation/pull/1616

#### Changes
* Encode legitimate content that includes SQL and script (URI-encode, then BASE64 encode)
* Move QWPDemo-related files into `simpletest` module
* Bump `@labkey/api`

### CODE REVIEW

File: query/webapp/query/QueryEditorPanel.js [hunk 1]
```diff
@@ -613,10 +613,10 @@ Ext4.define('LABKEY.query.QueryEditorPanel', {
         if (!this.query.userDefined || !this.query.canEdit)
             json.ff_queryText = null;
         else
-            json.ff_queryText = this.getSourceEditor().getValue();
+            json.ff_queryText = LABKEY.Utils.wafEncode(this.getSourceEditor().getValue());
 
         if (this.query.canEditMetadata)
-            json.ff_metadataText = this.getMetadataEditor().getValue();
+            json.ff_metadataText = LABKEY.Utils.wafEncode(this.getMetadataEditor().getValue());
```
Comment (labkey-nicka) [inline query/webapp/query/QueryEditorPanel.js:619]: There is another direct usage of the `query-saveSourceQuery` endpoint in [QWPDemoData.js](https://github.com/LabKey/platform/blob/34485139477676b3ae60b9eeee99cb184f898971/query/webapp/QWPDemoData.js#L74).

Comment (labkey-jeckels) [reply query/webapp/query/QueryEditorPanel.js:619]: Moved to `testAutomation` and updated to use `wafEncode()`

File: api/src/org/labkey/api/util/PageFlowUtil.java [hunk 2]
```diff
@@ -638,6 +640,51 @@ public static <T> Set<T> set(T... args)
...
+    // Text that is first urlencoded to flatten UNICODE to ASCII then base64 encoded to avoid WAF rejection
+    public static final String WAF_PREFIX = "/*{{base64/x-www-form-urlencoded/wafText}}*/";
+
+    /**
+     * De-obfuscates content that's often intercepted by web application firewalls that are scanning for likely
+     * SQL or script injection. We have a handful of endpoints that intentionally accept SQL or script, so we
+     * encode the text to avoid tripping alarms. It's a simple BASE64 encoding that obscures the content, and lets the
+     * WAF scan for and reject malicious content on all other parameters. See issue 48509.
+     */
+    public static String wafDecode(String encoded)
```
Comment (labkey-nicka) [inline api/src/org/labkey/api/util/PageFlowUtil.java]: nit: Annotate `wafDecode` and `wafEncode` as `@Nullable`.

**Review by labkey-nicka — APPROVED** [review]

File: api/src/org/labkey/api/util/PageFlowUtil.java (hunk 2 above)
Comment (labkey-jeckels) [reply api/src/org/labkey/api/util/PageFlowUtil.java]: Added

**Pull request merged**